### PR TITLE
site: Added release schedule page

### LINKED ISSUE #10409: site: Add release schedule to documentation
We release minikube once per month at the end of the month, with a beta prior. Let's add a more detailed release schedule so that people won't be surprised by a new release.

### PR DESCRIPTION
**Pull request by spowelljr**
Closes #10409

Screen Shot 2021-03-03 at 2 18 13 PM

### CODE REVIEW

File: site/content/en/docs/release/_index.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+---
```
Comment (medyagh) [inline site/content/en/docs/release/_index.md:1]: move this under contribution

**Review by medyagh — CHANGES_REQUESTED** [review]

File: site/content/en/docs/release/_index.md (hunk 1 above)
Comment (spowelljr) [reply site/content/en/docs/release/_index.md:1]: But if someone has no interest in contributing to the project, they're just looking for the release schedule I don't think they'd look for it under `Contribution`.

Comment (spowelljr) [reply site/content/en/docs/release/_index.md:1]: But I guess release schedule only really matters to contributers.

Comment (spowelljr) [reply site/content/en/docs/release/_index.md]: Done

**Review by medyagh — APPROVED** [review]

**Pull request merged**